### PR TITLE
[binn] new port

### PR DESCRIPTION
**Pull request by et-nik**
This adds the C++ Binn library https://github.com/liteserver/binn/tree/38019c11e582e5078436b8257887e79a33db8b28

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@et-nik, thanks for contributing to vcpkg!

Here is failed triplets:
```
Processing arm-uwp                        413 vs 413
    binn                                                        Fail  
Processing x64-uwp                        441 vs 441
    binn                                                        Fail 
```

**Comment by et-nik** [comment]
@PhoebeHui How I can fix? I can't view details: "401 Not Authorized"

**Comment by Rastaban** [comment]
Here are the results from the CI system:

```
Starting package 1/1: binn:x64-uwp
Building package binn[core]:x64-uwp... 
Could not locate cached archive: C:\vsts\_work\1\s\archives\ad\ad0fbff403a2a6be27ac2bb7f12d3c843686f86b.zip
-- Using cached C:/vsts/_work/1/s/downloads/liteserver-binn-38019c11e582e5078436b8257887e79a33db8b28.tar.gz
-- Extracting source C:/vsts/_work/1/s/downloads/liteserver-binn-38019c11e582e5078436b8257887e79a33db8b28.tar.gz
-- Using source at C:/vsts/_work/1/s/buildtrees/binn/src/9a33db8b28-abed577a29
-- Installing: C:/vsts/_work/1/s/packages/binn_x64-uwp/include/binn/binn.h
-- Configuring x64-uwp
-- Building x64-uwp-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:175 (message):
    Command failed: D:/downloads/tools/cmake-3.12.4-windows/cmake-3.12.4-win32-x86/bin/cmake.exe;--build;.;--config;Debug;--target;install;--;/p:VCPkgLocalAppDataDisabled=true;/p:UseIntelMKL=No
    Working Directory: C:/vsts/_work/1/s/buildtrees/binn/x64-uwp-dbg
    See logs for more information:
      C:\vsts\_work\1\s\buildtrees\binn\install-x64-uwp-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/binn/portfile.cmake:20 (vcpkg_install_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package binn:x64-uwp failed with: BUILD_FAILED
Elapsed time for package binn:x64-uwp: 23.82 s

```
[install-x64-uwp-dbg-out.log](https://github.com/Microsoft/vcpkg/files/2963089/install-x64-uwp-dbg-out.log)

Doing a quick scan of the logs it looks like you probably just need to suppress the warnings that are being treated as errors.  There are some libraries that just don't support UWP, if on further investigation you determine that this is the case then we will accept the new port with UWP failures.

**Comment by soroshsabz** [comment]
@Rastaban how to access VCPKG-PR-Eager log? why you can access it and @et-nik does not?

**Comment by PhoebeHui** [comment]
Arm-uwp and x64-uwp still failed in latest CL test results.

[install-x64-uwp-dbg-out.log](https://github.com/Microsoft/vcpkg/files/3002759/install-x64-uwp-dbg-out.log)

**Comment by grdowns** [comment]
Everything passes. Thanks for the new port, @et-nik!

File: ports/binn/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION 3.8)
+
+project(binn
+  VERSION 1.0
+  DESCRIPTION "Binn is a binary data serialization format designed to be compact, fast and easy to use."
+  HOMEPAGE_URL "https://github.com/liteserver/binn"
+)
+
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```
Comment (cenit) [inline ports/binn/CMakeLists.txt]: Didn’t we say “no” to this line if we had to write it? @Rastaban @grdowns

Comment (grdowns) [reply ports/binn/CMakeLists.txt]: That's right! Sorry this slipped through. Thanks for the catch, @cenit.

Comment (grdowns) [reply ports/binn/CMakeLists.txt]: Created issue #6097

**Pull request merged**